sci-libs/vtk/vtk-9.0.1.ebuild

```diff
@@ -0,0 +1,541 @@
+	openmp? (
+		|| (
+			sys-devel/gcc[openmp(+)]
+			sys-devel/clang-runtime[openmp(+)]
+		)
+	)
+"
+
+PATCHES=(
+	"${FILESDIR}"/${P}-0001-fix-kepler-compute_arch-if-CUDA-toolkit-11-is-used.patch
+	"${FILESDIR}"/${P}-0002-use-cmake-internal-auto-search-for-cuda-devices.patch
+	"${FILESDIR}"/${PN}-8.2.0-freetype-2.10.3-provide-FT_CALLBACK_DEF.patch
+)
+
+DOCS=( CONTRIBUTING.md README.md )
+
+CHECKREQS_DISK_BUILD="3G"
+
+pkg_pretend() {
+	if use examples; then
+		CHECKREQS_DISK_BUILD="4G"
+	fi
+	if use cuda; then
+		# NOTE: This should actually equal to (number of build jobs)*7G,
+		# as any of the cuda compile tasks can take up 7G!
+		# 10.2 GiB install directory, 6.4 GiB build directory with max. USE flags
+		CHECKREQS_MEMORY="7G"
+		CHECKREQS_DISK_BUILD="17G"
+	fi
+	check-reqs_pkg_setup
+}
+
+pkg_setup() {
+	if use examples; then
+		CHECKREQS_DISK_BUILD="4G"
+	fi
+	if use cuda; then
+		CHECKREQS_MEMORY="7G"
+		CHECKREQS_DISK_BUILD="17G"
+	fi
+	check-reqs_pkg_setup
+
+	use java && java-pkg-opt-2_pkg_setup
+	use python && python-single-r1_pkg_setup
+	use web && webapp_pkg_setup
+}
+
+src_prepare() {
+	# If we have system libraries available use these and delete
+	# the respecting files in ${S}/ThirdParty to save some space.
+	# Note: libharu is omitted: vtk needs an updated version (2.4.0)
+	# Note: no valid xdmf2 targets are found for system xdmf2
+	# Note: no valid target found for h5part and mpi4py
+	# TODO: diy2 exodusII h5part kissfft libharu verdict vpic
+	#	vtkm xdmf2 xdmf3 zfp
+	local -a DROPS=( doubleconversion eigen expat freetype gl2ps glew
+		hdf5 jpeg jsoncpp libproj libxml2 lz4 lzma netcdf ogg png pugixml
+		sqlite theora tiff utf8 zlib )
+	use pegtl && DROPS+=( pegtl )
+
+	local x
+	for x in ${DROPS[@]}; do
+		ebegin "Dropping bundled ${x}"
+		rm -r ThirdParty/${x}/vtk${x} || die
+		eend $?
+	done
+	unset x
+
+	if use doc; then
+		einfo "Removing .md5 files from documents."
+		rm -f "${WORKDIR}"/html/*.md5 || die "Failed to remove superfluous hashes"
+		sed -e "s|\${VTK_BINARY_DIR}/Utilities/Doxygen/doc|${WORKDIR}|" \
+			-i Utilities/Doxygen/CMakeLists.txt || die
+	fi
+
+	if use test; then
+		if use java; then
+			eapply "${FILESDIR}/${P}-0004-disable-failing-java-test.patch"
+		fi
+		if use examples; then
+			eapply "${FILESDIR}/${P}-0005-disable-failing-tests.patch"
+		fi
+	fi
+
+	cmake_src_prepare
+
+	if use cuda; then
+		cuda_add_sandbox -w
+		cuda_src_prepare
+	fi
+
+	if use test; then
+		ebegin "Copying data files to ${BUILD_DIR}"
+		mkdir -p "${BUILD_DIR}/ExternalData" || die
+		pushd "${BUILD_DIR}/ExternalData" >/dev/null || die
+		ln -sf ../../VTK-${PV}/.ExternalData/README.rst . || die
+		ln -sf ../../VTK-${PV}/.ExternalData/SHA512 . || die
+		popd >/dev/null || die
+#		dosym "${S}/.ExternalData/README.rst" "${BUILD_DIR}/ExternalData/README.rst"
+#		dosym "${S}/.ExternalData/SHA512" "${BUILD_DIR}/ExternalData/SHA512"
+#		dosym8 -r "${S}/.ExternalData/MD5" "${BUILD_DIR}/ExternalData/MD5"
+		eend "$?"
```

Comment:
Can someone please take a look at those `dosym` / `dosym8` commands? They don't work for me. I've not seen any error when they were run, but the files are not linked after they've run.

Reply:
@ulm Here's an example where dosym8 didn't work the way, I suspected it to work, like we have talked about at the Gentoo e.V. workshop.

Reply:
The problem is that `dosym` is an installation command, i.e., it is intended to be used in src_install(). It will expand the second path relative to `${ED}`, so your links are created in `${ED}${BUILD_DIR}/ExternalData/MD5` etc. I've just checked, the links are there, in the `image` directory. :smile:

So, the right way to do this in src_prepare() is with `ln -s` as you do in the lines above the commented-out ones.

Reply:
Thanks, I'm gonna remove the comments then :)